src/Controllers/LoginSessionController.php

```diff
@@ -29,42 +33,48 @@ class LoginSessionController extends Controller
      */
     public function remove(HTTPRequest $request): HTTPResponse
     {
-        return $this->removeLoginSession($request);
-    }
-
-    private function removeLoginSession(HTTPRequest $request): HTTPResponse
-    {
-        $failureMessage = _t(__CLASS__ . '.REMOVE_FAILURE', 'Something went wrong.');
-        try {
-            // Ensure CSRF protection
-            if (!SecurityToken::inst()->checkRequest($request)) {
-                return $this->jsonResponse([
+        if (empty(Security::getCurrentUser())) {
+            return $this->jsonResponse(
+                [
                     'success' => false,
-                    'message' => $failureMessage
-                ]);
-            }
+                    'message' => _t(
+                        __CLASS__ . '.YOUR_SESSION_HAS_EXPIRED',
+                        'Your session has expired'
+                    )
+                ],
+                401
+            );
+        }
 
-            $id = $request->param('ID');
-            $loginSession = LoginSession::get()->byID($id);
-            if (!$loginSession) {
-                return $this->jsonResponse([
+        // Validate CSRF token and ID parameter
+        $id = $request->param('ID');
+        if (!SecurityToken::inst()->checkRequest($request) || empty($id) || !is_numeric($id)) {
+            return $this->jsonResponse(
+                [
                     'success' => false,
-                    'message' => $failureMessage
-                ]);
-            }
+                    'message' => _t(
+                        __CLASS__ . '.INVALID_REQUEST',
+                        'Invalid request'
+                    )
+                ],
+                400
+            );
+        }
 
-            if (!$loginSession->canDelete()) {
-                $message = _t(__CLASS__ . '.REMOVE_PERMISSION', 'You do not have permission to delete this record.');
-                return $this->jsonResponse([
+        $loginSession = LoginSession::get()->byID($id);
+        if (!$loginSession || !$loginSession->canDelete()) {
+            // We're not making a difference between a non-existent session and session you can not revoke
+            // to prevent an adversary from scanning session IDs
+            return $this->jsonResponse(
+                [
                     'success' => false,
-                    'message' => $message
-                ]);
-            }
-        } catch (Exception $e) {
-            return $this->jsonResponse([
-                'success' => false,
-                'message' => $failureMessage
-            ]);
+                    'message' => _t(
+                        __CLASS__ . '.COULD_NOT_LOG_OUT_OF_THIS_SESSION',
+                        'Could not log out of this session. Maybe it has already been terminated.'
```

Comment:
You will get this message if you try to revoke a session that no longer exist. Typical scenario where you would get this is when you have already logout of the other device after the loading the "Authenticated devices" form field.

@silverstripeux Let me know if you want to tweak this wording

Reply:
Maybe this is enough?
This session could not found or is no longer active.

If the above is not enough then we could clarify but it might be too much text? 
It may have already been terminated or timed out.

Reply:
Updated the wording 
![image](https://user-images.githubusercontent.com/1168676/121979628-a5060100-cdde-11eb-985d-0775d5db148c.png)